### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/Tusky/commit/90d6f77204cf7ec84bc26fe39d1dcb898b205f9b#commitcomment-89134418

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/Tusky/commit/188229110fb725ede6a5538ef9d2b6015a77f85c
(note: I've skipped `much much` -- this repository is slightly playful)

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: app/src/main/res/values-ar/strings.xml [hunk 1]
```diff
@@ -342,10 +342,10 @@
...
     <string name="description_post_media_no_description_placeholder">مِن دون وصف    </string>
     <string name="description_post_reblogged">أعاد تدوينه    </string>
-    <string name="description_visiblity_public">للعامة    </string>
-    <string name="description_visiblity_unlisted">غير مُدرَج    </string>
-    <string name="description_visiblity_private">المتابِعون    </string>
-    <string name="description_visiblity_direct">مباشر    </string>
+    <string name="description_visibility_public">للعامة    </string>
+    <string name="description_visibility_unlisted">غير مُدرَج    </string>
+    <string name="description_visibility_private">المتابِعون    </string>
+    <string name="description_visibility_direct">مباشر    </string>
```
Comment (jsoref) [inline app/src/main/res/values-ar/strings.xml:348]: Some projects ask contributors to adjust localizations, others ask them not to. It's about 50/50. I've included the changes here, but it's trivial to drop them.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -507,16 +507,16 @@
     <string name="description_post_bookmarked">
         Bookmarked
     </string>
-    <string name="description_visiblity_public">
+    <string name="description_visibility_public">
```
Comment (jsoref) [inline app/src/main/res/values/strings.xml:510]: Either way, this is the change that corresponds to the code change.

File: app/src/test/java/com/keylesspalace/tusky/db/TimelineDaoTest.kt [hunk 3]
```diff
@@ -448,7 +448,7 @@ class TimelineDaoTest {
             favourited = !even,
             bookmarked = false,
             sensitive = even,
-            spoilerText = "spoier$statusId",
+            spoilerText = "spoiler$statusId",
```
Comment (jsoref) [inline app/src/test/java/com/keylesspalace/tusky/db/TimelineDaoTest.kt:451]: ??

**Review by jsoref — COMMENTED** [review]
Most corrections automatically suggested by Google Sheets. All fault mine (and `workaround` is my fault -- I regret it)

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt [hunk 4]
```diff
@@ -394,7 +394,7 @@ class ComposeActivity :
             updateVisibleCharactersLeft()
         }
 
-        // work around Android platform bug -> https://issuetracker.google.com/issues/67102093
+        // workaround Android platform bug -> https://issuetracker.google.com/issues/67102093
```
[the remark this answers is not in the document]
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt:397]: yes it is

File: app/src/test/java/com/keylesspalace/tusky/db/TimelineDaoTest.kt (hunk 3 above)
Comment (connyduck) [reply app/src/test/java/com/keylesspalace/tusky/db/TimelineDaoTest.kt:451]: its just some string to fill something into the content warning ("spoiler")

**Review by connyduck — APPROVED** [review]

**Pull request merged**